src/entry.rs

```diff
@@ -47,6 +47,58 @@ impl<'a, K: IntKey, V> Entry<'a, K, V> {
 
         (cache_ix, vals_ix)
     }
+
+    /// Ensures a value is in the entry by inserting the provided value if empty, and returns
+    /// a mutable reference to the value in the entry.
+    pub fn or_insert(self, default: V) -> &'a mut V {
+        match self {
+            Entry::Occupied(entry) => entry.into_mut(),
+            Entry::Vacant(entry) => entry.insert(default),
+        }
+    }
+
+    /// Ensures a value is in the entry by inserting the result of the provided function if empty,
+    /// and returns a mutable reference to the value in the entry.
+    pub fn or_insert_with<F>(self, default: F) -> &'a mut V
+    where
+        F: FnOnce() -> V,
+    {
+        match self {
+            Entry::Occupied(entry) => entry.into_mut(),
+            Entry::Vacant(entry) => entry.insert(default()),
+        }
+    }
+
+    /// Ensures a value is in the entry by inserting, if empty, the result of the provided function.
+    ///
+    /// Unlike its usefulness in standard hashmaps, since intmap keys are Copy this function is of little
+    /// particular use to intmaps but provided anyway to match the standard entry API
+    pub fn or_insert_with_key<F>(self, default: F) -> &'a mut V
+    where
+        F: FnOnce(&K) -> V,
```

Comment:
The key type is `Copy`, so no need to pass a reference here. When introducing the entry API, I intentionally chose to diverge from the standard API and pass the key by value everywhere. And I think the `IntMap` API is consistently using `K` instead of `&K`. E.g. see #70.

Reply:
As I state in its documentation, this function's functionality doesn't make much sense anyway if key is Copy. I just added it to make migrations completely seamless.

Reply:
If you still want it changed let me know, but might as well remove it then

Reply:
It still might make sense without the reference, e.g. if you pass the entry to another function and the function needs access to the key.

Currently the `IntMap` API doesn't try to replicate the std API 1:1 and diverges where it makes sense. That means that the user must do minor changes when migrating from std to `IntMap`. Based on this I would either remove the method or change it to pass-by-value. But I'm not the maintainer, maybe @JesperAxelsson has a different opinion.

Reply:
Thanks for the review @jakoschiko .
Both with and without reference each have an advantage. Still as no other function use a reference I agree with @jakoschiko here. The function does have value if someone just wants to evaluate if they want to use IntMap they can just deref.